vnext/Shared/Threading/BatchingQueueThread.cpp

```diff
@@ -14,7 +15,11 @@ BatchingQueueThread::BatchingQueueThread(
 BatchingQueueThread::~BatchingQueueThread() noexcept {}
 
 void BatchingQueueThread::runOnQueue(std::function<void()> &&func) noexcept {
-  ThreadCheck();
+  std::lock_guard<std::mutex> lck(m_mutex);
```

Comment:
Just checking: Is a mutex not a bit too heavy here in the queue processing and scheduling? Should we use an interlock compareexchange?